script/deploy/validate-deployment-logs.ts

```diff
@@ -0,0 +1,549 @@
+#!/usr/bin/env bun
+
+/**
+ * Deployment Logs Validation Script
+ *
+ * This script validates the integrity and consistency of deployment logs
+ * between MongoDB and the local JSON file. It helps ensure data quality
+ * during the migration from JSON-based to MongoDB-based logging.
+ *
+ * Features:
+ * - Compare record counts between MongoDB and JSON
+ * - Identify missing or extra records in either source
+ * - Detect data inconsistencies
+ * - Find duplicate entries
+ * - Validate data integrity (timestamps, addresses, etc.)
+ */
+
+import { readFileSync } from 'fs'
+import path from 'path'
+
+import { defineCommand, runMain } from 'citty'
+import { consola } from 'consola'
+
+import { getEnvVar } from '../demoScripts/utils/demoScriptHelpers'
+
+import {
+  DatabaseConnectionManager,
+  type IConfig,
+  type IDeploymentRecord,
+  RecordTransformer,
+  createDeploymentKey,
+} from './shared/mongo-log-utils'
+
+/**
+ * Helper to get environment variable with optional default
+ */
+const getEnvVarWithDefault = (
+  varName: string,
+  defaultValue: string
+): string => {
+  try {
+    return getEnvVar(varName)
+  } catch {
+    return defaultValue
+  }
+}
+
+/**
+ * Configuration for validation
+ */
+const config: IConfig = {
+  mongoUri: getEnvVarWithDefault('MONGODB_URI', 'mongodb://localhost:27017'),
```

Comment:
why do we default to a locally running mongoDB? we do not have this set up as far as I am aware?